app/src/main/java/com/leoleo/androidgithubsearch/di/ApiModule.kt

```diff
@@ -0,0 +1,56 @@
+package com.leoleo.androidgithubsearch.di
+
+import com.leoleo.androidgithubsearch.BuildConfig
+import com.leoleo.androidgithubsearch.data.api.GithubService
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+import java.util.concurrent.TimeUnit
+import javax.inject.Singleton
+
+@InstallIn(SingletonComponent::class)
+@Module
+internal object ApiModule {
+    private const val TIMEOUT_SEC: Long = 30
+
+    @Singleton
+    @Provides
+    fun provideGithubService(
+        okHttpClient: OkHttpClient,
+        moshi: Moshi,
+    ): GithubService {
+        return Retrofit.Builder()
+            .client(okHttpClient)
+            .baseUrl(BuildConfig.GITHUB_API_DOMAIN)
+            .addConverterFactory(MoshiConverterFactory.create(moshi))
+            .build()
+            .create(GithubService::class.java)
+    }
+
+    @Singleton
+    @Provides
+    fun provideOkHttpClient(debug: Boolean = BuildConfig.DEBUG): OkHttpClient {
+        val builder = OkHttpClient.Builder()
+            .connectTimeout(TIMEOUT_SEC, TimeUnit.SECONDS)
+            .readTimeout(TIMEOUT_SEC, TimeUnit.SECONDS)
+        if (debug) {
+            val httpLoggingInterceptor =
+                HttpLoggingInterceptor(HttpLoggingInterceptor.Logger.DEFAULT).apply {
+                    level = HttpLoggingInterceptor.Level.BODY
+                }
+            builder.addInterceptor(httpLoggingInterceptor)
+        }
+        return builder.build()
+    }
```

Comment:
`OkHttpClient`オブジェクトはシングルトンにするとパフォーマンスが上がる！
https://square.github.io/okhttp/4.x/okhttp/okhttp3/-ok-http-client/#okhttpclients-should-be-shared
```
OkHttp performs best when you create a single OkHttpClient instance and reuse it for all of your HTTP calls. This is because each client holds its own connection pool and thread pools. Reusing connections and threads reduces latency and saves memory. Conversely, creating a client for each request wastes resources on idle pools.
```